src/std_impls/collections.rs

```diff
@@ -154,64 +95,139 @@ unsafe impl<T: Finalize, S: BuildHasher> Finalize for HashSet<T, S> {
     }
 }
 
-// TUPLES
-unsafe impl<A, B> GcDeref for (A, B)
-where
-    A: GcDeref,
-    B: GcDeref,
-{
-}
-unsafe impl<A, B> GcDrop for (A, B)
-where
-    A: GcDrop,
-    B: GcDrop,
-{
-}
-unsafe impl<A, B> GcSafe for (A, B)
-where
-    A: GcSafe,
-    B: GcSafe,
-{
-}
+/// Vec like structure means that it implemented `Iter<T>`
+#[macro_export]
```

Comment:
I don't think we want to `macro_export` this. Much better to force either explicit implementation, or using the derive. Otherwise we introducing a macro that can generate potentially unsound code without the word `unsafe` in the name, which is an anti-pattern